src/main/java/com.company/backtracking/WordBreak.java

```diff
@@ -0,0 +1,95 @@
+package com.company.backtracking;
+
+import java.util.*;
+
+public class WordBreak {
+    private Trie trie;
+    private boolean wordBreak(String str, List<String> dict) {
+        trie = new Trie();
+        for(String word : dict) {
+            trie.insertTrie(word);
+        }
+        return solveWordBreak(str, 0);
+    }
+
+    private boolean solveWordBreak(String str, int index) {
```

Comment:
In terms of code quality, name of the boolean methods should start with `can`, `has`, `is`, `are` etc